torch/csrc/jit/backends/generate_debug_handles.cpp

```diff
@@ -0,0 +1,40 @@
+#include <torch/csrc/jit/backends/generate_debug_handles.h>
+
+#include <stack>
+
+namespace torch {
+namespace jit {
+
+NodeToDebugHandle generate_debug_handles(const std::shared_ptr<Graph>& graph) {
+  NodeToDebugHandle node_to_debug_handles;
+  BackendDebugHandleManager* dbg_handle_manager_ptr =
+      getBackendDebugHandleManager();
+
+  // Note now we make having a valid debug_handle_manager a must.
+  // This avoids silently failing when say some code change results in
+  // to_backend not creating appropriate debug_handle_manager to
+  // be used with backend's preprocess function.
+  TORCH_CHECK(
```

Comment:
Without changing current preprocess signature, this will be tricky. Plus for python binding it seems having generate_debug_handles as a separate function is easier?